pkg/controller/test-images-distributor/test_images_distributor.go

```diff
@@ -323,6 +333,100 @@ func (r *reconciler) ensureImagePullSecret(namespace string, client ctrlruntimec
 	return nil
 }
 
+const ciOperatorPullerRoleName = "ci-operator-image-puller"
+
+func referenceCIOperatorRole(namespace string) *rbacv1.Role {
+	return &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      ciOperatorPullerRoleName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"image.openshift.io"},
+				Resources: []string{"imagestreamtags", "imagestreams", "imagestreams/layers"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		},
+	}
+}
+
+func (r *reconciler) ensureCIOperatorRole(namespace string, client ctrlruntimeclient.Client, log *logrus.Entry) error {
+	reference := referenceCIOperatorRole(namespace)
+	existingRole := &rbacv1.Role{}
+	name := types.NamespacedName{Namespace: namespace, Name: reference.Name}
+	if err := client.Get(r.ctx, name, existingRole); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get role %s: %w", name.String(), err)
+		}
+		if err := client.Create(r.ctx, reference); err != nil {
+			return fmt.Errorf("failed to create role %s: %w", name.String(), err)
+		}
+		log.Infof("Successfully created %s role", reference.Name)
+		return nil
+	}
+	// Reset ObjectMeta to make comparable
+	rv := existingRole.ResourceVersion
+	existingRole.ObjectMeta = reference.ObjectMeta
+	if reflect.DeepEqual(existingRole, reference) {
```

Comment:
Shouldn't this be `semantic.Equality.DeepEqual()`?

Reply:
Yeah, this was a bit ad hoc. I've sanitized it to use common upsert code everywhere. PTAL.